packages/Confirmation/test/confirmation.spec.js

```diff
@@ -18,28 +18,31 @@ function renderComponent() {
 }
 
 describe("Confirmation", () => {
-  let getByText;
-  let confirmationComponent;
-  // const onOpenManageTableSidePanel = jest.fn();
-
-  beforeEach(() => {
-    ({ getByText } = renderComponent());
-    confirmationComponent = getByText(/confirm delete/i);
-  });
-
   it("should show confirmation panel by default when mounted ", () => {
-    expect(confirmationComponent).toBeVisible();
+    const { getByText } = renderComponent({ defaultIsOpen: true });
+    expect(getByText(/Confirm Delete/)).toBeVisible();
   });
 
   it("should trigger the confirm callback when confirm button is clicked", () => {
-    // expect(confirmationComponent).toBeVisible();
+    const onConfirmFunc = jest.fn();
+    const { getByText } = renderComponent({
+      onConfirm: onConfirmFunc,
+    });
+    fireEvent.click(getByText(/trigger/i));
+    fireEvent.click(getByText(/Confirm Delete/i));
+    expect(onConfirmFunc).toHaveBeenCalled();
   });
 
-  it("should close the confirmation when clicking outside of the confirmation panel", () => {
-    // expect(confirmationComponent).toBeVisible();
-  });
-
-  it("should trigger the onConfirm callback when clicking the confirm button", () => {
-    // expect(confirmationComponent).toBeVisible();
-  });
+  // todo  - fix why onCloseFunc not being called
+  // it("should close the confirmation when clicking confirm", async () => {
+  //   const onCloseFunc = jest.fn();
+  //   const { getByText } = renderComponent({
+  //     onClose: onCloseFunc,
+  //   });
+  //   fireEvent.click(getByText(/trigger/i));
+  //   fireEvent.click(getByText(/Confirm Delete/i));
+  //   await wait(() => {
+  //     expect(onCloseFunc).toHaveBeenCalled();
+  //   });
+  // });
```

Comment:
Unsure why the `onCloseFunc` callback isn't being detected as being called here. Intend to revisit if cannot find solution. Have tested in the browser and is working for me there.

Reply:
Can you get this test working please (otherwise it will never get done).

Reply:
maybe wrapped on window.requestAnimationFrame(()=> { 
  // run your test
})

Reply:
tried `async` `await`, `window.requestAnimationFrame` and 'setTimeout` but i'm finding none result in the `onClose` being being called... 🤔